test/test_binary.py

```diff
@@ -158,53 +158,53 @@ def test_uuid_subtype_4(self):
     def test_legacy_java_uuid(self):
         # Test decoding
         data = self.java_data
-        docs = bson.decode_all(data, CodecOptions(SON, False, PYTHON_LEGACY))
+        docs = bson.decode_all(data, CodecOptions(SON, False, PYTHON_LEGACY))  # type: ignore[type-var]
```

Comment:
what about:
```
SON = SON[str, Any]
...
```
or 
```
son = SON[str, Any]
...CodecOptions(son,...
...
```
Would that remove the need for `# type:` comments?

Reply:
I had thought it wouldn't work on Python 3.6, but it looks like it does, updating.

Reply:
Okay, the actual issue was that you can't use `issubclass` on a generic alias in Python 3.10 and PyPy, I added a workaround.